### PR TITLE
[MNT] Upgrade prophet to >=1.1 and remove pystan from all_extras dependencies

### PR DESCRIPTION
**Pull request by khrapovs**
#### Reference Issues/PRs

Closes https://github.com/alan-turing-institute/sktime/issues/2886

#### What does this implement/fix? Explain your changes.

Upgrade `prophet` to `>=1.1` and remove `pystan` from `all_extras` dependencies.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [X] I've added unit tests and made sure they pass locally.
- [X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

### CODE REVIEW

**Comment by aiwalter** [comment]
@khrapovs nice, was just about to open a PR as well :) If this works, do you mind to also slightly update the installation docs about `prophet` and `pystan?`
https://github.com/alan-turing-institute/sktime/blob/adad11df6da0a133ba75d5cbe88f0c92dc774889/docs/source/installation.rst#L75-L98

**Comment by khrapovs** [comment]
> @khrapovs nice, was just about to open a PR as well :) If this works, do you mind to also slightly update the installation docs about `prophet` and `pystan?`
> 
> https://github.com/alan-turing-institute/sktime/blob/adad11df6da0a133ba75d5cbe88f0c92dc774889/docs/source/installation.rst#L75-L98

Sorry, I was faster :D
Sure. I'll wait for the tests and then push the doc update.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Excellent!!!

I think we need to have a quick discussion on whether the dep changes are significant enough to make this 0.13.0, i.e., to be put in the mid July release

**Comment by aiwalter** [comment]
@khrapovs we need to also update the `.yml` files, you can find them by searching for `pystan`. This dedicated installation commands can go now also

**Comment by khrapovs** [comment]
> @khrapovs we need to also update the `.yml` files, you can find them by searching for `pystan`. This dedicated installation commands can go now also

Now I am in uncharted territory. I have removed some conda lines. Since many years I do not use conda, so I have no idea if what I did it 100% correct. Please, be very careful reviewing this.

**Comment by aiwalter** [comment]
looks good now :) thx

File: docs/source/installation.rst
Comment (aiwalter) [inline docs/source/installation.rst:191]: do we still need this?

Comment (khrapovs) [reply docs/source/installation.rst:191]: No idea. I was wondering too. Why is it here in the first place?

Comment (aiwalter) [reply docs/source/installation.rst:191]: I think its was there bcz of the 3 commands before. So as thexy are gone it can go also I think

Comment (aiwalter) [reply docs/source/installation.rst:191]: @fkiraly might know

Comment (khrapovs) [reply docs/source/installation.rst]: I have removed this step completely. Let me know if it should be kept and in which wording.

Comment (aiwalter) [reply docs/source/installation.rst]: yes should be fine as it was also not in the `.yaml` file.

**Review by aiwalter — DISMISSED** [review]

**Comment by fkiraly** [comment]
@lmmentel, @khrapovs, any danger via the removed conda lines?

> I have removed some conda lines. Since many years I do not use conda, so I have no idea if what I did is 100% correct. Please, be very careful reviewing this.

**Comment by khrapovs** [comment]
> @lmmentel, @khrapovs, any danger via the removed conda lines?

If you trust the test suit, 0% danger. If you don't, what is missing? ;)

**Comment by lmmentel** [comment]
Looks good @khrapovs, great work :raised_hands:

File: pyproject.toml
Comment (aiwalter) [inline pyproject.toml:55]:
```suggestion
    "cmdstanpy<1.0.2", # can be relaxed if issue is resolved at prophet or cmdstanpy side
```

**Review by lmmentel — APPROVED** [review]

**Comment by tcuongd** [comment]
> btw @tcuongd we are facing issues with the `prophet==1.1` in combination with the yesterday released `cmdstanpy==1.0.2`. I cant exactly give you a reproducible code but some of our unit tests were failing after `cmdstanpy` released yesterday.

Oh dang I'll take a look, I'll ask Brian the cmdstanpy maintainer for help as well. Thanks!

**Comment by khrapovs** [comment]
> Oh dang I'll take a look, I'll ask Brian the cmdstanpy maintainer for help as well. Thanks!

@tcuongd Please, tell us of any outcome of your investigation here: https://github.com/alan-turing-institute/sktime/issues/2898. Thanks!

**Comment by fkiraly** [comment]
> @fkiraly Could you please approve this PR

There you see @khrapovs, not all PR need my approval to be merged!

**Pull request merged**
